src/Agent.Plugins/GitSourceProvider.cs

```diff
@@ -844,12 +848,13 @@ public string GenerateAuthHeader(AgentTaskPluginExecutionContext executionContex
             cancellationToken.ThrowIfCancellationRequested();
             executionContext.Progress(80, "Starting checkout...");
             string sourcesToBuild;
-
+            executionContext.Debug($"refFetchedByCommit : {refFetchedByCommit}");
+            
             if (refFetchedByCommit != null)
             {
                 sourcesToBuild = refFetchedByCommit;
             }
-            else if (IsPullRequest(sourceBranch) || string.IsNullOrEmpty(sourceVersion))
+            else if (IsPullRequest(sourceBranch) && string.IsNullOrEmpty(sourceVersion))
```

Comment:
Could you explain in a few words how this fixes the regression since the change you linked already has this logic OR logic too? I'm just trying to understand why Julio's change broke things and this fix will go back to the expected behavior

Reply:
With Julio's change, any PR will pass the condition without ever checking whether the sourceVersion is available for the checkout or not. Once the condition passes sourceToBuild used for checkout gets set to the branch ref. With this change, we are also checking if the sourceVersion is not available for a PR before setting the sourceToBuild to branch ref. If sourceVersion is not null then we set the sourceToBuild to the sourceVersion (aka SHA) in the else condition.